frontend/src/main/java/com/google/tripmeout/frontend/servlet/TripParentServlet.java

```diff
@@ -35,35 +35,37 @@ public TripParentServlet(TripStorage storage, Gson gson) {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String userId = getUserId();
-    response.setContentType("application/json");
-    response.setStatus(HttpServletResponse.SC_OK);
     PrintWriter writer = response.getWriter();
-    writer.print(gson.toJson(storage.getAllUserTrips(userId)));
-    writer.flush();
-    writer.close();
+    try {
+      String userId = getUserId();
+      response.setContentType("application/json");
+      response.setStatus(HttpServletResponse.SC_OK);
+      writer.print(gson.toJson(storage.getAllUserTrips(userId)));
+    } finally {
+      writer.close();
```

Comment:
You might want to wrap all `writer.close()` calls in a try-catch block to prevent the exception from escaping (depending on how strict we want to be about leaking details to clients). I'll leave that decision up to you.